functions/pipe/auto_compact.py

```
@@ -0,0 +1,9957 @@
+        return
+    for source in sources:
+        if not isinstance(source, dict):
+            continue
+        source_info = source.get("source", {})
+        if not isinstance(source_info, dict):
+            continue
+        if not (source_info.get("name", "") or source_info.get("id", "")):
+            continue
+        try:
+            await event_emitter({"type": "source", "data": source})
+        except Exception:
```

Ruff (0.15.20)

[warning] Do not catch blind exception: `Exception`

(BLE001)